code/modules/chemistry/tools/robospray.dm

```diff
@@ -69,19 +87,33 @@
 		"<span class='notice'>You inject [amt_prop] units of [propername]. [src] now contains [botreagents[currentreagent] - amt_prop] units.</span>")
 		logTheThing("combat", user, M, "uses a cybernetic hypospray to inject [constructTarget(M,"combat")] with [amt_prop] units of [propername] at [log_loc(user)].")
 
-		M.reagents.add_reagent(botreagents[picker], amt_prop)
+		M.reagents.add_reagent(currentreagent, amt_prop)
 		botreagents[currentreagent] = botreagents[currentreagent] - amt_prop
 		tooltip_rebuild = 1
+		update_icon()
 		playsound(get_turf(M), src.sound_inject, 80, 0)
 		return 0
 
 	process()
 		..()
+		var/refill_amount = (ticker.round_elapsed_ticks - last_refill) / 29
+		if (last_refill < 0)
+			refill_amount = 1
+		last_refill = ticker.round_elapsed_ticks
+
+		refill_amount += extra_refill
+		extra_refill = refill_amount - round(refill_amount)
+		refill_amount = round(refill_amount)
```

Comment:
Essentially, it's just to make sure the hypospray always has an integer amount of each chemical in it to make the messages say 1 unit rather than 1.0023 units.